src/server/game/Globals/ObjectAccessor.cpp

```diff
@@ -81,6 +81,41 @@ HashMapHolder<Player>::MapType const& ObjectAccessor::GetPlayers()
 template class TC_GAME_API HashMapHolder<Player>;
 template class TC_GAME_API HashMapHolder<Transport>;
 
+void PlayerNameMapHolder::Insert(Player* p)
+{
+    boost::unique_lock<boost::shared_mutex> lock(*GetLock());
+
+    GetContainer()[p->GetName()] = p;
+}
+
+void PlayerNameMapHolder::Remove(Player* p)
+{
+    boost::unique_lock<boost::shared_mutex> lock(*GetLock());
+
+    GetContainer().erase(p->GetName());
+}
+
+// note that input string shall be normalized by the caller
```

Comment:
I don't like this requirement - you can easily normalize the name inside (pass by value), before acquiring the lock